views/pulsar/v2/helpers/form.html.twig

```diff
@@ -2107,3 +2107,87 @@ data-*        | string | Data attributes, eg: `'data-foo': 'bar'`
+                    'required': options.required|default(false)
+                })
+            }}
+            <div class="controls">
+                <div class="table-container">
+                    <table class="table table--full repeatable__table">
+                        <thead class="repeater__preview-headings">
+                            <tr>
+                                {% for heading in options.headings %}
+                                    <th data-repeater-for-name="{{ heading.name }}">{{ heading.label }}</th>
+                                {% endfor %}
+                                <th rowspan="1" colspan="1"></th>
+                            </tr>
+                        </thead>
+                        <tbody class="repeater__preview-data">
+                            <tr class="repeater__empty-placeholder">
+                                <td colspan="{{ headingLength }}" class="muted">{{ options.empty }}</td>
+                                <td class="table__td table__td--align-right table-row__actions"></td>
+                            </tr>
+                        </tbody>
+                        <tfoot>
+                        <tr class="repeater__group" data-repeater-new-group>
+                            <td class="table__form" colspan={{ headingLength + 1 }}>
+
+{% endspaceless %}
+{% endmacro %}
+
+{% macro repeatable_end() %}
+{% import '@pulsar/pulsar/v2/helpers/html.html.twig' as html %}
+{% spaceless %}
+                                <div class="form__actions repeater__group-actions">
+                                    {{
+                                        html.button({
+                                            'data-repeater-save-group': 'true',
+                                            'label': 'Save',
+                                            'class': 'btn--primary repeater__save-group',
+                                            'type': 'button'
+                                        })
+                                    }}
+
+                                    {{
+                                        html.button({
+                                            'data-repeater-cancel-save': 'true',
+                                            'label': 'Cancel',
+                                            'class': 'btn--naked repeater__cancel-group',
+                                            'type': 'button'
+                                        })
+                                    }}
+                                </div>
+                            </td>
+                        </tr>
+                        </tfoot>
+                    </table>
+                </div>
+                {{
+                    html.button({
+                        'data-repeater-add-group': 'true',
+                        'label': 'Add',
+                        'class': 'btn--primary repeater__add-group',
+                        'type': 'button'
```

Comment:
Don't need type here, should default to button

Reply:
Oh I see, i didn't check the default type, although I set it explicitly because it was submitting my forms, I'll double check that default value.

Reply:
More investigation, we add `type="button"` as a default option when we define a button as an input, however if we want a button to be a `<button/>` we don't set `type="button"` as a default (which I think is correct as it might be odd if a button *didn't* submit a form by default), however in this case I don't want that behaviour.

Reply:
👍